devdoc/WorkUnits.rst

```diff
@@ -430,12 +430,47 @@ Once all of the required workflow items are executed, the workunit is marked as
 executing dependencies and then executing that particular item.
 
 Most of the interesting work in an ECL query is done within a graph.  The call ctx->executeGraph will either execute the graph locally
 (in the case of hthor and roxie), or add the workunit onto a queue (for Thor).  Whichever happens, control will pass to that engine.
 
+Specialised Workflow Items
+=================
+Each item mode/type can affect the dependency structure of the workflow:
+
+- Sequential/Ordered
+  An item is made to contain all of the actions in the sequential/ordered statement. Each action becomes a dependency of this item (unless there is no item for the action). The dependencies, and therefore the actions, must be executed in order.
+  For sequential, an extra item is always inserted between the sequential item and each of its dependencies. This means that if other sequential statements reference the same attribute, it will only be performed once.
+
+- Persist
+  When the persist workflow service is used, two items are created. The item that contains the expression graphs, stores the other item's wfid.
+  This second item is used to determine whether the persist is up to date.
+
+- Condition (IF)
+  The IF function has either 2 or 3 arguments: the expression, the trueresult, and sometimes the falseresult. These arguments translated into workflow items and are stored as the dependencies of a condition item.
+  They are always stored in order.
+
+- Contingency (SUCCESS/FAILURE)
+  When a contingency clause is defined for an expression, the expression item stores a wfid for the contingency. The contingency items are not dependencies of the expression.
+  If both success and failure are used, then the item will store the wfid of each contingency. The contingency action is translated into items, just like the larger query.
+
+- Recovery
+  When a workflow item fails, if it has a recovery clause it will be re-executed. The clause contains actions defined by the programmer to try to fix the problem.
+  The clause is stored differently to SUCCESS/FAILURE, in that the recovery clause is a dependency of the expression.
+  In order to stop the recovery clause from being executed like other dependencies, it is marked with WFStateSkip.
+
+- Independent
+  This specifier is used when a programmer wants to common up code for the query. It prevents the same piece of code from being executed twice in different contexts.
+  To achieve this, an extra item is inserted between the expression and whichever items depend on it. This means that although the attribute can be referenced several times, it will only be executed once.
+
+Whilst this feature relates to persist/independent, it doesn't affect the type of the workflow Item:
+- Cluster name
+  If a persist/independent action should be executed on a different cluster, the generated code calls a factory function to update a workunit variable. Hence, this feature has no effect on the workflow xml.
```

Comment:
I'm not sure the last lione is worth including, since I suspect this may change soon.